app/models/ont/plate_with_samples_factory.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+# Ont namespace
+module Ont
+  # PlateWithSamplesFactory
+  # A factory for bulk inserting a plate and all of its dependents
+
+  # rubocop:disable Metrics/ClassLength
+  class PlateWithSamplesFactory
+    include ActiveModel::Model
+
+    validate :check_plate_factory
+
+    def initialize(attributes = {})
+      @attributes = attributes
+      time = DateTime.now
+      @timestamps = { created_at: time, updated_at: time }
+    end
+
+    def process
+      @plate_factory = Ont::PlateFactory.new(attributes)
+    end
+
+    def save(**options)
+      return false unless options[:validate] == false || valid?
+
+      @serialised_plate_data = plate_factory.bulk_insert_serialise(self, validate: false)
+      bulk_insert
+    end
+
+    # Serialisation
+
+    def ont_request_data(ont_request, tag_id)
+      {
+        ont_request: serialise_ont_request(ont_request),
+        tag_id: tag_id
+      }
+    end
+
+    def well_data(well, request_data)
+      { well: serialise_well(well), request_data: request_data }
+    end
+
+    def plate_data(plate, well_data)
+      { plate: serialise_plate(plate), well_data: well_data }
+    end
+
+    private
+
+    attr_reader :attributes, :timestamps, :plate_factory, :serialised_plate_data
+
+    def serialise_ont_request(ont_request)
+      {
+        uuid: SecureRandom.uuid,
+        external_id: ont_request.external_id,
+        name: ont_request.name
+      }.merge(timestamps)
+    end
+
+    def serialise_well(well)
+      { position: well.position }.merge(timestamps)
+    end
+
+    def serialise_plate(plate)
+      { barcode: plate.barcode }.merge(timestamps)
+    end
+
+    # Saving and Validation
+
+    def check_plate_factory
+      return if plate_factory.valid?
+
+      plate_factory.errors.each do |k, v|
+        errors.add(k, v)
+      end
+    end
+
+    def bulk_insert
+      plate = false
+      ActiveRecord::Base.transaction do
+        plate = insert_plate
+        request_uuids = insert_wells_and_ont_requests(plate.id)
+        request_ids_by_uuid = get_request_ids_by_uuid(request_uuids)
+        well_ids_by_position = get_well_ids_by_position(plate.wells)
+        insert_joins(well_ids_by_position, request_ids_by_uuid)
+      end
+      plate
+    rescue StandardError => e
+      errors.add('import was not successful:', e.message)
+      false
+    end
+
+    def insert_plate
+      plate_data = serialised_plate_data[:plate]
+      Plate.insert_all!([plate_data])
+      Plate.find_by!(barcode: plate_data[:barcode])
+    end
+
+    def insert_wells_and_ont_requests(plate_id)
+      requests_data = []
+      wells_data = serialised_plate_data[:well_data].map do |well_data|
+        requests_data.concat(well_data[:request_data].map { |req_data| req_data[:ont_request] })
+        well_data[:well].merge({ plate_id: plate_id })
+      end
```

Comment:
Rather than creating wells with the map and appending requests as a separate action inside the map (which is a small abuse of the `map` function) it would probably have been better to output arrays of two items.  The first is the well data and the second is an array of requests for that well.  So you'd get something like:
```
parsed_data = [
  [well1_data, [well1_req1, well1_req2]],
  [well2_data, [well2_req1, well2_req2]]
]
```
Then you `map` the first values out to `wells_data` and `flat_map` the `requests_data` out from the second values.  Something like:
```
wells_data = parsed_data.map(&:first)
requests_data = parsed_data.flat_map(&:last)
```